akka-remote/src/main/scala/akka/remote/transport/netty/CustomTrustManagerFactory.scala

```diff
@@ -0,0 +1,7 @@
+package akka.remote.transport.netty
+
+import javax.net.ssl.TrustManager
+
+trait CustomTrustManagerFactory {
+  def create(filename: String, password: String): Array[TrustManager]
+}
```

Comment:
I wasn't sure this would be easy to implement in Java but it is:
```
package akka.remote.transport.netty;

import javax.net.ssl.TrustManager;

public class TestCustomTrustManagerFactory implements CustomTrustManagerFactory {
  @Override
  public TrustManager[] create(String filename, String password) {
    return new TrustManager[0];
  }
}
```
Perhaps neat to add a test that checks this remains the case?

Reply:
Yes, for sure. I'm trying to wrap my head around scalatest and your tcs at the moment. I have a faint memory of some other reflection thingie being tested for java interoperabillity but of course can't find it now ;)

Reply:
I changed it so there is one instantiation mechanism. No special cases to test. One approach with a default and possibility to specialise.